android/modules/network/src/java/ti/modules/titanium/network/TiHTTPClient.java

```diff
@@ -791,8 +791,23 @@ public void clearCookies(String url)
 	public void setRequestHeader(String header, String value)
 	{
 		if (readyState <= READY_STATE_OPENED) {
-			headers.put(header, value);
-
+			if (value == null) {
+				// If value is null, remove header
+				headers.remove(header);
```

Comment:
The cookies are being (can be) set in the headers.

Also the null value is following this behavior: https://github.com/appcelerator/APSHTTPClient/blob/c8531ff81a381c90ad445ceff891ded99214e6a9/APSHTTPClient/APSHTTPRequest.m#L172

https://github.com/appcelerator/titanium_mobile/blob/master/apidoc/Titanium/Network/HTTPClient.yml#L249

If I misunderstood it somehow, do let me know.